### PR TITLE
Np 46501 viewingScope & excludeSubunits

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search2/ticket/TicketQuery.java [hunk 1]
```diff
@@ -129,7 +135,7 @@ public TicketQuery withFilterCurrentUser(String userName) {
         this.currentUser = userName;
         if (isUserOnly(ticketTypes)) {
             final var viewOwnerOnly = new TermQueryBuilder(OWNER_USERNAME, userName)
-                .queryName(OWNER.asCamelCase());
+                .queryName(OWNER.asCamelCase() + POST_FILTER);
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/ticket/TicketQuery.java:138]: bugfix

File: search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java [hunk 2]
```diff
@@ -25,7 +25,7 @@ public void set(QueryBuilder... filters) {
     }
 
     public void add(QueryBuilder builder) {
-        this.filters.removeIf(filter -> filter.getName().equals(builder.getName()));
+        this.filters.removeIf(filter -> filter.queryName().equals(builder.queryName()));
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java:28]: bugfix

File: search-commons/src/main/java/no/unit/nva/search2/ticket/TicketQuery.java [hunk 3]
```diff
@@ -164,7 +173,7 @@ public TicketQuery withFilterTicketType(TicketType... ticketTypes) {
         this.ticketTypes = ticketTypes.clone();
         final var filter =
             new TermsQueryBuilder(TYPE_KEYWORD, Arrays.stream(ticketTypes).map(TicketType::toString).toList())
-                .queryName(TICKETS + TYPE);
+                .queryName(TicketParameter.TYPE.asCamelCase() + POST_FILTER);
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/ticket/TicketQuery.java:176]: bugfix

File: search-commons/src/test/resources/mapping_test_tickets.json [hunk 4]
```diff
@@ -16,14 +16,6 @@
...
       "include_in_parent": true
     },
-    "organizations": {
-      "properties": {
-        "partOf": {
-          "type": "nested",
-          "include_in_parent": true
-        }
-      }
-    },
```
Comment (StigNorland) [inline search-commons/src/test/resources/mapping_test_tickets.json:26]: this isn't used, `organizations ` should have been `organization`

but it doesn't matter , default mapping works well

File: search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java [hunk 5]
```diff
@@ -25,7 +25,7 @@ public void set(QueryBuilder... filters) {
     }
 
     public void add(QueryBuilder builder) {
-        this.filters.removeIf(filter -> filter.getName().equals(builder.getName()));
```
Comment (sondrev) [inline search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java:28]: If filters needs to be unique for each queryName you could make filter a HashMap with queryName as key

**Review by sondrev — COMMENTED** [review]
There are no tests for your new code?

File: search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java (hunk 5 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java:28]: good point

**Pull request merged**